### PR TITLE
community: support bind_tools for ChatMlflow

### PR DESCRIPTION
**Pull request by serena-ruan**
Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- **Description:** 
Support ChatMlflow.bind_tools method
Tested in Databricks:
image

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Comment by serena-ruan** [comment]
cc @prithvikannan

**Comment by BenWilson2** [comment]
This implementation looks great! 
Can we add some tests to validate tool submission check logic and tool type parsing validation to ensure that changes in agent handlers in the core library doesn't create any breaking changes with this added agent handling capability?

**Comment by serena-ruan** [comment]
@baskaryan Could I get a review on this PR?

File: libs/community/langchain_community/chat_models/mlflow.py
Comment (B-Step62) [inline libs/community/langchain_community/chat_models/mlflow.py:1]: These `_convert_xxx` functions are copied from `ChatOpenAI`, I have a few concerns on doing so:
https://github.com/langchain-ai/langchain/blob/59880a9147501160c6c336159cacfe62515b0b7e/libs/partners/openai/langchain_openai/chat_models/base.py#L95

* How do we maintain these logic compatible with the original implementation?
* Is it guaranteed that all Databricks served model to have the same format for tool request/response? Have you test this with existing FMAPIs?
* If this model only works on endpoints that strictly conforms to OpenAI spec, should we consider subclassing `BaseChatOpenAI`? It seems that the difference is just what URL to hit, similar to AzureOpenAI. For reference [Databricks](https://github.com/run-llama/llama_index/blob/d15f45bcec205a4cd0c8f2a4b7c829385e2910e1/llama-index-integrations/llms/llama-index-llms-databricks/llama_index/llms/databricks/base.py#L7) subclasses OpenAILike class (equivalent to BaseChatOpenAI) in LlamaIndex.

**Review by baskaryan — APPROVED** [review]

File: libs/community/langchain_community/chat_models/mlflow.py
Comment (B-Step62) [reply libs/community/langchain_community/chat_models/mlflow.py:1]: Oh, I didn't know the goal of this PR is Llama-3-70B alone... Can we add a warning in document mention that the tool functionality only guaranteed to work with Llama3(or FMAPI if it works with DBRX, etc)? 

> Maybe yes if we have such assumption, we need to double check with dev team on this. But I think it's out of the scope of this PR as this PR aims to support bind_tools function.

Ok, I will scope it as a part of package migration. My comment is about maintenance concern from the retrospective learning.

File: docs/docs/integrations/tools/databricks.ipynb [hunk 1]
```diff
@@ -47,9 +47,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from langchain_openai import ChatOpenAI\n",
```
Comment (B-Step62) [inline docs/docs/integrations/tools/databricks.ipynb:50]: Can we remove `langchain_openai` from requirement?

**Review by B-Step62 — COMMENTED** [review]
Can you also update https://python.langchain.com/v0.2/docs/integrations/chat/databricks/#model-features?

**Review by B-Step62 — APPROVED** [review]

**Review by B-Step62 — APPROVED** [review]
LGTM! Thank you for adding the documentation and references!

**Comment by serena-ruan** [comment]
@baskaryan Could you help me merge this PR? Thanks!

**Comment by serena-ruan** [comment]
@jacoblee93 @efriis Could you help merging this PR? Thank you!

**Pull request merged**